app/models/job.ts

```diff
@@ -514,11 +587,22 @@ export class Job extends Record implements JobRecord {
    */
   constructor(fields: JobRecord) {
     super(fields);
-    this.updateStatus(fields.status || JobStatus.ACCEPTED, fields.message);
+    let initialMessage: string;
+    try {
+      // newer jobs will have stringified JSON stored in the database
+      this.statesToMessages = JSON.parse(fields.message);
+      initialMessage = this.message;
+    } catch (e) {
```

Comment:
When using this pattern a lot of times we like to look for a specific exception type rather than catching any exception. I'm not sure if there's a specific exception to expect in this case.

Reply:
✅